DataFormats/TrackReco/interface/TrackExtraBase.h

```diff
@@ -51,8 +51,13 @@ class TrackExtraBase
     }
 
     /// get a ref to i-th recHit
-    TrackingRecHitRef recHitRef(unsigned int i) const {                                                               
-        return TrackingRecHitRef(m_hitCollection,m_firstHit+i);
+    TrackingRecHitRef recHitRef(unsigned int i) const {
+      if(m_hitCollection.productPtr()) {
+        TrackingRecHitRef::finder_type finder;
+        TrackingRecHitRef::value_type const* item = finder(*(static_cast<TrackingRecHitRef::product_type const*>(m_hitCollection.productPtr())), m_firstHit+i);
+        return TrackingRecHitRef(m_hitCollection.id(), item, m_firstHit+i);
+      }
+      return TrackingRecHitRef(m_hitCollection,m_firstHit+i);

```

Comment:
Ref has changed. The cached pointer it contains used to point at the container but now it points at the element in the container (much more similar to the way Ptr works and this allows Ptr to Ref conversions). The cached pointer in the RefProd continues to point at the container and m_hitCollection has a RefCore from a RefProd, so the pointer needs to be converted from pointing to the container to pointing at the element in this case.